src/Algebra/Action/Structures.agda

```diff
@@ -0,0 +1,95 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Structure of the Action of one (pre-)Setoid on another
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Relation.Binary.Core using (Rel)
+
+module Algebra.Action.Structures
+  {c a ℓ r} {M : Set c} {A : Set a} (_≈ᴹ_ : Rel M ℓ) (_≈_ : Rel A r)
+  where
+
+open import Data.List.Base
+  using (List; []; _∷_; _++_; foldl; foldr)
+open import Data.List.NonEmpty.Base
+  using (List⁺; _∷_)
+open import Data.List.Relation.Binary.Pointwise as Pointwise
+  using (Pointwise; []; _∷_)
+open import Function.Base using (id)
+open import Level using (Level; _⊔_)
+
+open import Algebra.Action.Definitions _≈ᴹ_ _≈_
+
+private
+  variable
+    x y z : A
+    m n p : M
+    ms ns ps : List M
+
+
+------------------------------------------------------------------------
+-- Basic definitions: fix notation
+
+record IsLeftAction : Set (a ⊔ r ⊔ c ⊔ ℓ) where
```

Comment:
Hmm is there a reason why the hierarchy in `Structures` is not the same as the hierarchy in `Bundles`? e.g. `Bundles` talks about setoids and monoids which don't feature here?